client/allocrunner/taskrunner/artifact_hook.go

```diff
@@ -52,7 +53,18 @@ func (h *artifactHook) Prestart(ctx context.Context, req *interfaces.TaskPrestar
 
 		h.logger.Debug("downloading artifact", "artifact", artifact.GetterSource)
 		//XXX add ctx to GetArtifact to allow cancelling long downloads
-		if err := getter.GetArtifact(req.TaskEnv, artifact, req.TaskDir.Dir); err != nil {
+		clientEnv := make(map[string]string, len(req.TaskEnv.EnvMap))
+		for k, v := range req.TaskEnv.EnvMap {
+			clientEnv[k] = v
+		}
+		clientEnv[taskenv.AllocDir] = req.TaskDir.SharedAllocDir
+		clientEnv[taskenv.TaskLocalDir] = req.TaskDir.Dir
+		clientTaskEnv := taskenv.NewTaskEnv(
+			clientEnv,
+			req.TaskEnv.DeviceEnv(),
+			req.TaskEnv.NodeAttrs,
+		)
```

Comment:
not happy with this deconstruction, but it was either that or change a lot of plumbing to get the  `taskEnv.Builder` into here.

Reply:
in hindsight, i'm thinking that a cleaner abstraction is to pass around a single `TaskEnv`/`EnvReplacer` with two methods: `ReplaceEnv` and `ReplaceEnvClient`. Could even augment it with all of the interpolation/join/sandbox-check logic and clean this stuff up.

Will take a pass at that now.